src/basilisp/core.lpy

```diff
@@ -2515,26 +2515,8 @@
   [x]
   (fn [& ^:no-warn-when-unused args] x))
 
+;; redefined later
 (defn reduce
```

Comment:
Internally Clojure just uses a different private function (`reduce1` although I'd prefer something like `reduce*` which matches the naming convention for other private-ish functions).

I don't like simply re-`def`-ing.

Reply:
Done.